### ISSUE TITLE
Training loop update

### ISSUE THREAD

**Issue by jindrahelcl** [ufal/neuralmonkey#671]
Here are a few suggestions for improving the training user experience:

1. During logging, execute `run_on_dataset` __before__ execution of trainer.
2. Execute it also after that and report the loss/eval differences.
3. Report the scores from before and after update and their difference in the log and in TensorBoard.
4. Optionally, introduce verbosity control to the logging so the compact version of the log is preserved.
